### DISCUSSION TITLE
Show analytics and statistics

### DISCUSSION THREAD

**Discussion by brutella** [brutella/hkknx-public#241] [Ideas]
I'm thinking about a new feature to get more insights about your smart home. Something like statistics about how many telegrams are received from the bus, or how many telegrams are sent by hkknx to the bus. 

This might be useful to see bottlenecks or problems which would otherwise get overlooked.

Are there any other ideas which kind of statistics are useful?

[a comment by an automated account is not shown]

**Comment by mirkolenz** [comment]
Great idea! I would be very interested in statistics like "average telegrams per second" or a ranking of the "most used telegrams". This would make it much easier to adjust telegram frequencies correctly (e.g., of wind or temperature sensors).

**Comment by brutella** [comment]
There are now charts for KNX bus load, cpu load, memory utilization and internal storage usage under *Status* in version [2.8.1-b1](https://github.com/brutella/hkknx-public/releases/tag/2.8.1-b1). Looking for feedback. 👀

**Reply by mirkolenz** [reply]
Great, thanks for working on this. I am extremely busy at the moment and would rather not use a beta version. Do you maybe have a screenshot of the new features?

**Reply by brutella** [reply]
<img width="1138" alt="Screenshot 2024-02-09 at 19 59 11" src="https://github.com/brutella/hkknx-public/assets/125641/10a84595-80ec-4d85-a081-478cd6a5a25c">

**Reply by mirkolenz** [reply]
Looks nice and clean! I especially like the ability to monitor the bus load (something I wanted to have for a long time actually). Based on the screenshots, I have two questions:

1. Is the CPU load monitored permanently in the background? If so, what is the performance impact?
2. You show the bus load as a percentage. Is there a standard upper limit of how many telegrams a bus can handle per second or does that depend on the hardware used (maybe newer bus devices can handle more load?)? Maybe it would be a nice to add the upper limit to the description?

Thanks!

Edit: Another question just came up:

3. What time scale is used on the x-axis? You already include a scale on the y-axis, so adding that information to the other axis as well would make it easier to interpret the data.

**Reply by brutella** [reply]
1. No, the statistics are gathered on demand. Once you close the page, the monitoring is stopped. (Maybe the KNX bus load will/can be monitored in the future.)
2. There will be a description of the bus load calculation. In short the maximum number of telegrams for a KNX bus depends on the length of the bus cable. With a max cable of 1,000 meters, the maximum number of telegrams is 28 telegrams per second. In practice this is more like 48 telegrams per second. hkknx can only observe the real KNX bus load, if it is directly connected to the bus. KNX tunnelling doesn't give you access to all telegrams on the bus. So this graphs doesn't really show the real bus load. But it's still interesting to see imo.
3. The x-scale is 2 minutes.

**Reply by mirkolenz** [reply]
Thanks for the clarifications. I think for the KNX bus load it may indeed be nice if the data is collected in the background. Also nice to know that the telegram frequency depends on the cable length (which makes sense). Will be interesting to test this out in my setup. Once I had the time to do so (hopefully at the end of February) I will let you know about any observations. Keep up your work!
